modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -639,7 +639,7 @@ class TaskProcessor {
         // -- map the inputs to a map and use to delegate closure values interpolation
         final secondPass = [:]
         int count = makeTaskContextStage1(task, secondPass, values)
-        makeTaskContextStage2(task, secondPass, count)
+        final batch = makeTaskContextStage2(task, secondPass, count)
```

Comment:
Let's rename to `foreignFiles` here and in the producing function

Reply:
Done